components/hab-spider/src/main.rs

```diff
@@ -22,38 +22,53 @@ extern crate walkdir;
 extern crate habitat_core as hab_core;
 extern crate builder_core as bldr_core;
 extern crate habitat_builder_protocol as protocol;
+extern crate habitat_builder_db as db;
+extern crate habitat_net as hab_net;
 extern crate clap;
+extern crate postgres;
+extern crate protobuf;
+extern crate r2d2;
+
+pub mod data_store;
+pub mod error;
 
 use std::io::{self, Write};
 use clap::{Arg, App};
 use time::PreciseTime;
 use bldr_core::package_graph::PackageGraph;
-use bldr_core::file_walker::FileWalker;
+//use bldr_core::file_walker::FileWalker;
+use data_store::DataStore;
 
 fn main() {
     env_logger::init().unwrap();
 
     let matches = App::new("hab-spider")
         .version("0.1.0")
         .about("Habitat package graph builder")
-        .arg(Arg::with_name("PATH").help("The path to the packages root").required(true).index(1))
+        .arg(Arg::with_name("URL").help("The DB connection URL").required(true).index(1))
```

Comment:
Yes, I did consider that, however it looked like it would be quite a bit more complicated to go that route and would need to add new APIs that the scheduler service doesn't actually need to be exposing.